src/osd/modules/netdev/pcap.cpp

```diff
@@ -189,15 +189,14 @@ netdev_pcap::netdev_pcap(const char *name, class device_network_interface *ifdev
 
 void netdev_pcap::set_mac(const char *mac)
 {
-	char filter[256];
 	struct bpf_program fp;
 	if(!m_p) return;
 #ifdef SDLMAME_MACOSX
-	sprintf(filter, "not ether src %.2X:%.2X:%.2X:%.2X:%.2X:%.2X and (ether dst %.2X:%.2X:%.2X:%.2X:%.2X:%.2X or ether multicast or ether broadcast or ether dst 09:00:07:ff:ff:ff)", (unsigned char)mac[0], (unsigned char)mac[1], (unsigned char)mac[2],(unsigned char)mac[3], (unsigned char)mac[4], (unsigned char)mac[5], (unsigned char)mac[0], (unsigned char)mac[1], (unsigned char)mac[2],(unsigned char)mac[3], (unsigned char)mac[4], (unsigned char)mac[5]);
+	auto filter = util::string_format("not ether src %2X:%2X:%2X:%2X:%2X:%2X and (ether dst %2X:%2X:%2X:%2X:%2X:%2X or ether multicast or ether broadcast or ether dst 09:00:07:ff:ff:ff)", (unsigned char)mac[0], (unsigned char)mac[1], (unsigned char)mac[2],(unsigned char)mac[3], (unsigned char)mac[4], (unsigned char)mac[5], (unsigned char)mac[0], (unsigned char)mac[1], (unsigned char)mac[2],(unsigned char)mac[3], (unsigned char)mac[4], (unsigned char)mac[5]);
```

Comment:
I can't see these strings in any language files, so don't think removing the '.'s breaks any translations.

Reply:
You should replace the `.` with `0` – it’s building a BPF filter expression.  It simply won’t work if you use `%2X` like that because it will produce spaces.

Reply:
Ah good spot. I didn't test below 0x10, sorry. Fixed